api/envoy/api/v2/route/route.proto

```diff
@@ -409,7 +421,17 @@ message CorsPolicy {
   // Specifies regex patterns that match allowed origins.
   //
   // An origin is allowed if either allow_origin or allow_origin_regex match.
-  repeated string allow_origin_regex = 8 [(validate.rules).repeated .items.string.max_bytes = 1024];
+  //
+  // .. attention::
+  //   This field has been deprecated in favor of `allow_origin_safe_regex` as it is not safe for
+  //   use with untrusted input in all cases.
+  repeated string allow_origin_regex = 8
+      [(validate.rules).repeated .items.string.max_bytes = 1024, deprecated = true];
+
+  // Specifies regex patterns that match allowed origins.
+  //
+  // An origin is allowed if either allow_origin or allow_origin_safe_regex match.
+  repeated type.matcher.RegexMatcher allow_origin_safe_regex = 11;
```

Comment:
given we already have API churn here, WDYT about deprecating both `allow_origin{,_regex}` and use StringMatcher? same apply to other places that StringMatcher can be used.

Reply:
I thought about this and did try to do this elsewhere, but there is a special case in CORS in terms of what '*' means so it's not super trivial. @dschaller WDYT?

Reply:
It should be ok to use a StringMatcher here. The wildcard is valid as either a pattern or a string so even if someone uses an exact string of `*` browsers should be alright.

Reply:
OK thanks I will switch and tag you on the review.